pdns/dnsname.hh

```diff
@@ -97,9 +98,8 @@ public:
   }
   DNSName(const DNSName& a) = default;
   DNSName(DNSName&& a) = default;
-  explicit DNSName(const char* p): DNSName(p, std::strlen(p)) {} //!< Constructs from a human formatted, escaped presentation
-  explicit DNSName(const char* p, size_t len);      //!< Constructs from a human formatted, escaped presentation
-  explicit DNSName(const std::string& str) : DNSName(str.c_str(), str.length()) {}; //!< Constructs from a human formatted, escaped presentation
+
+  explicit DNSName(const std::string_view& sw); //!< Constructs from a human formatted, escaped presentation
```

Comment:
I'm not sure it makes sense in that case to pass views by reference instead of passing them by value. Views are cheap to create and copy, and if I'm not mistaken requiring a reference here means we can only pass lvalues.

Reply:
Anything larger than a pointer is "large" imo :-). I'll revert that part.

Reply:
Thanks it was instructive, I went for a ref due to my Rust background like ` &str` (slice type on string) but since string_view are "like slice type" i should have passed it directly.